### ISSUE TITLE
Tyler's comments 2/7/17

### ISSUE THREAD

**Issue by tyleransom** [ModUDuke/causalinference#1]
Here are my comments from @BrianAronson's prototype, in chronological order:

1. Looking at all of the coding exercises, the answers appear to be in the comments for all of them (not sure if this is for the SVPs at eBay, or if I'm doing something wrong in DataCamp?) Below is an example of the code I submitted for the very first coding exercise (which ended up passing the error checking): 
```
PhoneSpecifications #Dataframe

print(Solution1<-"Luminanceeeeee"  ) #Write column name of specification with the largest ratio between the WePhone 10S and the Universe S10

PhoneSpecifications[1,]/PhoneSpecifications[2,]
Solution1 <- "Luminance (cd/m2)"

print(Solution2<-  1.4   ) #Write quotient of phone specifications selected in Solution1
Solution2 <- PhoneSpecifications[1,ncol(PhoneSpecifications)]/PhoneSpecifications[2,ncol(PhoneSpecifications)]
```
2. It would be good to have some transition going into the OHIE preliminary exercise. Something like "Now let's look at some results of a randomized experiment that studied the effect of having health insurance on health outcomes. The dataset in memory contains group-level observations from the study." Similarly, I'm not sure if we want videos prefacing this beforehand, or if the current setup has enough background info for the user.
3. We need to specify 95% confidence level for the last question of "Practice Reading Tables"

Other than that, I think things are looking good!

**Comment by BrianAronson** [comment]
Hey Tyler,

Thanks for your feedback. I agree with all of your suggestions and will implement them ASAP. 

With #1, were you logged into Jim's account, or your own? I think DataCamp remembers the answers that you previously submit to questions, and I had answered them on Jim's account to make sure there weren't any errors. If not, dang... I''ll have to fix that.

Also relating to #1, we should discuss how restrictive we want the answers to be on acceptable answers. As you noticed, "Luminanceeeeee" is right now coded as an acceptable answer for Solution 1 on that question. It's coded to accept any answer that begins with the string "[L/l]uminance." I did this so that users could answer the problem by hand and run into less spelling issues, but there are a bunch of drawbacks to this method:

1) Goofy answers like "luminanceeeeee" are acceptable.
2) The way that DataCamp's error check runs for multiple acceptable solutions, the solution needs to be printed in the console (hence all the "print()" functions), which is definitely not ideal.
3) The way that DataCamp's error check runs for multiple acceptable solutions, as long as the user printed the correct solution at some point (even if it is not in his final answer), it will be accepted as correct. So if a user printed "1.4 in the console" for solution 2, but changed his answer to  "5," it would still be marked as correct.

Given these drawbacks, I am now leaning toward restricting each question to only accept one answer, rather than keep it as it is now. Do you agree?

**Comment by tyleransom** [comment]
Thanks, Brian. I was logged in as Jim, which why I saw all the answers. I just tried it using my profile, and they were all blank.

I'm not sure what the best approach is for error checking. If we limit to only one acceptable answer, then users might get irritated about how "picky" the system is. On the other hand, if we don't limit the acceptable answers, then potentially wrong answers will pass through. So I'm not sure what the best thing is to do right now, other than not spending too much time on it when the structure of the exercises might change dramatically.
